yt/frontends/gadget/io.py

```diff
@@ -105,7 +106,13 @@ def _get_smoothing_length(self, data_file, position_dtype, position_shape):
         with h5py.File(data_file.filename, "r") as f:
             pcount = f["/Header"].attrs["NumPart_ThisFile"][ind].astype("int")
             pcount = np.clip(pcount - si, 0, ei - si)
-            ds = f[ptype]["SmoothingLength"][si:ei,...]
+            if self._dataset_type == "arepo_hdf5":
```

Comment:
I wonder if it's perhaps better to make an `IOHandlerArepoHDF5` class that inherits from `IOHandlerGadgetHDF5` rather than to have this class do double duty. If functionality continues to evolve for either frontend, this could become more difficult to maintain.